### PR TITLE
Skip IPv6 LB test in the k/s e2e

### PR DESCRIPTION
**Pull request by hakman**
Ref:
* https://github.com/kubernetes/kubernetes/pull/106992

### CODE REVIEW

**Comment by hakman** [comment]
```
[BeforeEach] [sig-apps] Deployment
  /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/framework.go:185
STEP: Creating a kubernetes client
Dec 13 12:56:00.634: INFO: >>> kubeConfig: /root/.kube/config
STEP: Building a namespace api object, basename deployment
STEP: Waiting for a default service account to be provisioned in namespace
[BeforeEach] [sig-apps] Deployment
  /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/apps/deployment.go:89
[It] should not disrupt a cloud load-balancer's connectivity during rollout
  /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/apps/deployment.go:161
Dec 13 12:56:00.957: INFO: Not supported for IPv6 clusters and providers [aws] (found aws)
[AfterEach] [sig-apps] Deployment
  /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/apps/deployment.go:83
Dec 13 12:56:01.021: INFO: Log out all the ReplicaSets if there is no deployment created
[AfterEach] [sig-apps] Deployment
  /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/framework.go:186
Dec 13 12:56:01.086: INFO: Waiting up to 3m0s for all (but 0) nodes to be ready
STEP: Destroying namespace "deployment-7546" for this suite.

[SKIPPING] [0.591 seconds]
[sig-apps] Deployment
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/apps/framework.go:23
  should not disrupt a cloud load-balancer's connectivity during rollout [It]
  /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/apps/deployment.go:161

  Not supported for IPv6 clusters and providers [aws] (found aws)
```

/cc @rifelpet

**Comment by rifelpet** [comment]
I think we would still need this when testing ipv6 on k8s 1.23

**Comment by hakman** [comment]
> I think we would still need this when testing ipv6 on k8s 1.23

Is it using the "master" kubetest for that?

**Comment by rifelpet** [comment]
yes, all of our periodic e2e jobs use the master branch to build kubetest2-kops: https://github.com/kubernetes/test-infra/blob/4684b5ffc0af4f0af0ebf324e51dde621ade7801/config/jobs/kubernetes/kops/templates/periodic.yaml.jinja#L15-L20

**Comment by hakman** [comment]
@rifelpet Should be good to go now.

File: tests/e2e/pkg/tester/skip_regex.go [hunk 1]
```diff
@@ -80,9 +80,8 @@ func (t *Tester) setSkipRegexFlag() error {
 		skipRegex += "|In-tree.Volumes.\\[Driver:.gcepd\\].*topology.should.provision.a.volume.and.schedule.a.pod.with.AllowedTopologies"
 	}
 
-	if cluster.Spec.CloudProvider == "aws" && utils.IsIPv6CIDR(cluster.Spec.NonMasqueradeCIDR) {
-		// AWS VPC Classic ELBs are IPv4 only
-		// ref: https://docs.aws.amazon.com/elasticloadbalancing/latest/classic/elb-internet-facing-load-balancers.html#internet-facing-ip-addresses
+	if strings.Contains(cluster.Spec.KubernetesVersion, "v1.23.0") && cluster.Spec.CloudProvider == "aws" && utils.IsIPv6CIDR(cluster.Spec.NonMasqueradeCIDR) {
```
Comment (rifelpet) [inline tests/e2e/pkg/tester/skip_regex.go:83]:
```suggestion
	if strings.Contains(cluster.Spec.KubernetesVersion, "v1.23.") && cluster.Spec.CloudProvider == "aws" && utils.IsIPv6CIDR(cluster.Spec.NonMasqueradeCIDR) {
```

Comment (hakman) [reply tests/e2e/pkg/tester/skip_regex.go]: Done

**Review by rifelpet — APPROVED** [review]

**Pull request merged**